lib/screens_web/views/v2/audio/reconstructed_alert_fullscreen_view.ex

```diff
@@ -0,0 +1,206 @@
+        unaffected_routes: unaffected_routes
+      }) do
+    ~E|<%= get_line_name(route_svg_names) %> trains are skipping this station<%= render_cause(cause) %>. Please seek an alternate route. <%= get_line_name(unaffected_routes) %> trains are stopping here as usual.|
+  end
+
+  # Case 3: Single station, single line impacted
+  # The pattern to match is very simple here, because other station closures will
+  # be caught by previous clauses, so this pattern is specifically for a closure here
+  def render_alert(%{
+        effect: :station_closure,
+        routes: route_svg_names,
+        cause: cause
+      }) do
+    ~E|This station is closed<%= render_cause(cause) %>. Please seek an alternate route. <%= get_line_name(route_svg_names) %> trains are skipping this station.|
+  end
+
+  # Shuttle here
+  def render_alert(
+        %{
+          effect: :shuttle,
+          routes: route_svg_names,
+          endpoints: endpoints,
+          cause: cause
+        } = alert
+      ) do
+    if Map.has_key?(alert, :is_transfer_station) do
+      ~E|There are no <%= get_line_name(route_svg_names) %> trains. Please use the shuttle bus. Shuttle buses are replacing <%= get_line_name(route_svg_names) %> trains between <%= elem(endpoints, 0) %> and <%= elem(endpoints, 1) %><%= render_cause(cause) %>. All shuttle buses are accessible.|
+    else
+      ~E|This station is closed. Please use the shuttle bus. Shuttle buses are replacing <%= get_line_name(route_svg_names) %> trains between <%= elem(endpoints, 0) %> and <%= elem(endpoints, 1) %><%= render_cause(cause) %>. All shuttle buses are accessible.|
+    end
+  end
+
+  def render_alert(
+        %{
+          effect: :suspension,
+          routes: route_svg_names,
+          endpoints: endpoints,
+          cause: cause
+        } = alert
+      ) do
+    if Map.has_key?(alert, :is_transfer_station) do
+      ~E|There are no <%= get_line_name(route_svg_names) %> trains<%= render_cause(cause) %>. Please seek an alternate route. Please note that there are no <%= get_line_name(route_svg_names) %> trains between <%= elem(endpoints, 0) %> and <%= elem(endpoints, 1) %>.|
+    else
+      ~E|This station is closed<%= render_cause(cause) %>. Please seek an alternate route. Please note that there are no <%= get_line_name(route_svg_names) %> trains between <%= elem(endpoints, 0) %> and <%= elem(endpoints, 1) %>.|
+    end
+  end
+
+  defp get_line_name([%{color: _color, text: _text, type: _type} | _tail] = routes) do
+    routes
+    |> Enum.map(fn route -> route.text end)
+    |> Util.format_name_list_to_string_audio()
+  end
+
+  defp get_line_name(routes) do
+    route_ids =
+      routes
+      |> Enum.map(fn route -> route.route_id end)
+
+    lines_with_branches =
+      route_ids
+      |> Enum.filter(&String.contains?(&1, "-"))
+      |> Enum.map(fn "Green-" <> branch_letter -> branch_letter end)
+
+    lines_without_branches =
+      route_ids
+      |> Enum.filter(&(!String.contains?(&1, "-")))
```

Comment:
`Enum.reject` might be a little easier to follow here, `Enum.reject(&String.contains?(&1, "-"))`